api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -15,11 +15,27 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // gRPC-JSON transcoder :ref:`configuration overview <config_http_filters_grpc_json_transcoder>`.
 // [#extension: envoy.filters.http.grpc_json_transcoder]
 
-// [#next-free-field: 10]
+// [#next-free-field: 11]
 message GrpcJsonTranscoder {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.transcoder.v2.GrpcJsonTranscoder";
 
+  enum UrlUnescapeSpec {
+    // URL path parameters will not decode RFC 6570 reserved characters.
+    // For example, segment `%2f%23/%20%2523` is unescaped to `%2f%23/ %23`.
+    ALL_CHARACTERS_EXCEPT_RESERVED = 0;
+
+    // URL path parameters will be fully URI-decoded except in
+    // cases of single segment matches in reserved expansion, where "%2F" will be
+    // left encoded.
+    // For example, segment `%2f%23/%20%2523` is unescaped to `%2f#/ %23`.
+    ALL_CHARACTERS_EXCEPT_SLASH = 1;
+
+    // URL path parameters will be fully URI-decoded.
+    // For example, segment `%2f%23/%20%2523` is unescaped to `/#/ %23`.
```

Comment:
How does this interact with Envoy's existing path normalization controls? https://github.com/envoyproxy/envoy/blob/4d77fc802c3bc1c517e66c54e9c9507ed7ae8d9b/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto#L493 

Keep in mind that we'e planning on making some changes here in the future https://github.com/envoyproxy/envoy/issues/6589

Reply:
`UrlUnescapeSpec` controls decoding logic only, it does not try to normalize path. The result of this decoding is never used in envoy itself to reconstruct back to the `:path` header.
So said, this logic is applied after envoy's path normalization and does honor it.

Reply:
I think there may be overlap in the future with path normalization, but I can see how these would be used distinctly in a transcoder.